File: intel-hex-to-bin.java/src/main/java/demo/IntelHexToBin.java
package demo;

import java.io.IOException;

import static com.google.common.base.Preconditions.checkArgument;

/**
 * Main application class.
 */
final public class IntelHexToBin {

    public static void main(String[] args) throws IOException {
        checkArgument(args.length != 0);

        String hexFileName = args[0];
        String binFileName = FileName.getBinFileName(args[0]);

        Converter converter = new Converter((byte) 0xFF);
        converter.convert(hexFileName, binFileName);
    }

}

File: intel-hex-to-bin.kotlin/src/main/kotlin/demo/IntelHexToBin.kt
package demo

import com.google.common.base.Preconditions.checkArgument
import java.io.IOException

/**
 * Main application object.
 */
object IntelHexToBin {

    @Throws(IOException::class)
    @JvmStatic fun main(args: Array<String>) {
        checkArgument(args.size != 0)

        val hexFileName = args[0]
        val binFileName = getBinFileName(args[0])

        val converter = Converter(0xFF.toShort())
        converter.convert(hexFileName, binFileName)
    }

}
